share/spice/hackage/hackage.js

```diff
@@ -0,0 +1,94 @@
+// curl https://www.haskell.org/hoogle/\?mode\=json\&hoogle\=json
+// API response format.
+// {"results" => [
+//    {"location"=>"http://hackage.haskell.org/package/JSON-Combinator",
+//     "self"=>"package JSON-Combinator",
+//     "docs"=> "A combinator library on top of a generalised JSON type.\n\nVersion 0.2.8 "}]}
+
+// get documentation string
+function docString(doc) {
+    return doc.replace(/Version .*/,"").trim();
+}
+
+// get version from docs.
+// supported version formats: [1.2, 1.2.3, 1.2.3.4]
+function version(doc){
+    var version = doc.match(/\d\.\d\.?\d?\.?\d?/);
+    return version === null ? undefined : version[0];
+}
+
+// converts "1.2.3" to 1230
+// converts "0.0.1" to 10
+// helps in comparing version strings in different format.
+function versionInt(v) {
```

Comment:
Whoops, these private functions need to be inside our closure (`(function(env) { ... } ()`), but outside our callback (`env.ddg_spice_hackage`).

Let's move them down, so they aren't declared in the global namespace.